doc/source/development/rfc/rfc94_field_precision_width_metadata.rst

```diff
@@ -0,0 +1,173 @@
+.. _rfc-94:
+
+=============================================================
+RFC 94: Numeric fields width/precision metadata
+=============================================================
+
+============== =============================================
+Author:        Alessandro Pasotti
+Contact:       elpaso @ itopen.it
+Started:       2023-Feb-17
+Status:        Development
+Target:        GDAL 3.7
+============== =============================================
+
+Summary
+-------
+
+The document proposes and describes the introduction of a new method
+to the `OGRFieldDefn` class that returns information for numeric real
+fields about how the precision and width have been calculated by the
+driver and have to be interpreted.
+
+Motivation
+----------
+
+Applications (for example OGR/GDAL utils and QGIS) may require information
+about the field width and precision in order to convert between different
+formats or to validate user input on editing.
+
+Different drivers may calculate the width and precision differently by including
+or not the decimal separator and/or the minus sign into the calculated length.
+
+There is currently no way for an application to access this information and this
+can lead to loss of width or precision while converting between formats or when the
+application decides to stay on the safe side and reduce the width reported by GDAL.
+
+Additionally, attention must be paid to the meaning of "width" and "precision":
+OGR "width" corresponds to SQL "precision" and OGR "precision" corresponds to SQL "scale".
+
+
+For reference:
+
+- https://trac.osgeo.org/gdal/ticket/6960
+- https://issues.qgis.org/issues/11755
+- https://issues.qgis.org/issues/15188#note-8
+- https://github.com/qgis/QGIS/issues/51849
+
+
+Current drivers behavior
+------------------------
+
+Here is a list of the drivers and how they behave with respect to the width and precision,
+(for databases we refer to the `NUMERIC` data type).
+
+================== ================================== =====================
+ Driver             Width Includes Decimal Separator   Width Includes Sign
+================== ================================== =====================
+ Shapefile          YES                                YES
+ MapInfo            ?                                  ?
+ PostgreSQL         NO                                 NO
+ MySQL              NO                                 NO
+ MSSQL              NO                                 NO
+ OCI                NO                                 NO
+ GPKG               N/A                                N/A
+ CSV (from .csvt)   N/A                                N/A
+ HANA               ?                                  ?
+ FlatGeoBuf         ?                                  ?
+ FileGDB            ?                                  ?
+ GML                ?                                  ?
```

Comment:
https://www.oreilly.com/library/view/xml-schema/0596002521/re52.html
"xs:totalDigits defines the maximum number of digits of decimal and derived datatypes (both after and before the decimal point, not counting the decimal point itself). "

https://www.oreilly.com/library/view/xml-schema/0596002521/re24.html
xs:fractionDigits defines the maximum number of fractional digits (i.e., digits that are after the decimal point) of an xs:decimal datatype. 

So it seems to be NO NO

and experimenting with the result of above test.csv converted to GML with "ogr2ogr test.gml test.csv", and then editing test.gml:
```
$ ogrinfo GMLAS:test.gml -oo VALIDATE=YES
ERROR 1: test.gml:19:31 value '1.234' has '3' fraction digits which exceeds fractionDigits facet value '2'
$ ogrinfo GMLAS:test.gml -oo VALIDATE=YES
ERROR 1: test.gml:13:34 value '-1123.45' has '6' total digits which exceeds totalDigits facet value '5'
```
confirms it